Author: Pino Toscano <pino@debian.org>
Description: Fix format in printf-like functions, like qWarning(...)
 Switch a qWarning(...) call to the stream-like version to avoid using
 a raw C string as printf-like format (triggering a compiling error with
 -Werror=format-security).
 .
 The issue has been fixed in Qt 4.8.
Last-Update: 2012-03-05
Forwarded: no
Origin: vendor

--- a/src/declarative/debugger/qdeclarativedebugserver.cpp
+++ b/src/declarative/debugger/qdeclarativedebugserver.cpp
@@ -207,9 +207,8 @@
                 }
 
             } else {
-                qWarning(QString::fromAscii("QDeclarativeDebugServer: Ignoring \"-qmljsdebugger=%1\". "
-                                            "Format is -qmljsdebugger=port:<port>[,block]").arg(
-                             appD->qmljsDebugArgumentsString()).toAscii().constData());
+                qWarning() << QString::fromAscii("QDeclarativeDebugServer: Ignoring \"-qmljsdebugger=%1\". "
+                                                 "Format is -qmljsdebugger=port:<port>[,block]").arg(appD->qmljsDebugArgumentsString());
             }
         }
 #else
